sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala

```diff
@@ -316,20 +346,45 @@ class RocksDBFileManager(
     }
   }
 
+  // Get latest snapshot version <= version
+  def getLatestSnapshotVersionAndUniqueId(
+      version: Long, checkpointUniqueId: Option[String] = None): Array[(Long, Option[String])] = {
+    val path = new Path(dfsRootDir)
+    if (fm.exists(path)) {
+      val versionAndUniqueIds = fm.list(path, onlyZipFiles)
+        .map(_.getPath.getName.stripSuffix(".zip").split("_"))
+        .filter {
+          case Array(ver, _) => ver.toLong <= version
+          case Array(ver) => ver.toLong <= version
+        }
+        .map {
+          case Array(version, uniqueId) => (version.toLong, Option(uniqueId))
+          case Array(version) => (version.toLong, None)
```

Comment:
We should not return this one for V1.

If this function is reused in V1 and V2, we should separate it, in my opinion. We should not return V1 and V2 files together. It is too confusing.

Reply:
Separated